lib/Compose.php

```diff
@@ -798,9 +798,11 @@ public function buildAndSendMessage(
         $headers = $this->_prepareHeaders($header, $opts);
 
         /* Add a Received header for the hop from browser to server. */
-        $headers->addHeaderOb(
-            Horde_Core_Mime_Headers_Received::createHordeHop()
-        );
+        if ($conf['compose']['add_received_header']) {
```

Comment:
Maybe `if ($conf['compose']['add_received_header'] ?? true)` is better?

That would preserve the current behavior, if no config changes are made.
- conf not set -> add the header (no change for existing configs)
- conf set to `null` -> add the header (this is admittedly a bit unintuitive)
- conf set to any truthy value -> add the header
- conf set to any falsy value -> do not set the header

Reply:
Well, I was only imitating the style for the rest of the section, which of course had fixed assumptions that settings exist. I was under the impression that adding the setting in `conf.xml` with a default gave it an explicit default value?

Reply:
The XML is only involved in the config UI. When you try to update/change configs through the UI, it will work.

You can actually have a config file with missing values and the XML will be ignored. Best practice would be to have the code behave as before when the value is missing.

Reply:
```suggestion
        if (!isset($conf['compose']['add_received_header']) || ($conf['compose']['add_received_header'] === true)) {
```
@midahp @towo what do you think? This should allow us to close this issue. 
We want to check for actual truth ("Yes, I want to opt in for the previous, potentially unsecure behaviour"), not for truthy somethings.
We want to cover missing config values / upgrade use cases and not break them.

Reply:
sure, that makes sense.